packages/app/ui/components/ChatOptionsSheet.tsx

```diff
@@ -673,6 +679,112 @@ export function ChannelOptionsSheetContent({
     onPressChatDetails({ type: 'group', id: group.id });
   }, [group, onPressChatDetails]);
 
+  const handleSummarizeChannel = useCallback(
```

Comment:
again, most of the business logic here should be encapsulated rather than inlined in the Sheet itself